src/storage/src/source/mysql/replication/events.rs

```diff
@@ -121,12 +121,13 @@ pub(super) async fn handle_query_event(
                 {
                     trace!(%id, "timely-{worker_id} DDL change \
                            dropped table: {dropped_table:?}: {err:?}");
-                    let (output_index, _) = &ctx.table_info[&dropped_table];
-                    let gtid_cap = ctx.data_cap_set.delayed(new_gtid);
-                    ctx.data_output
-                        .give(&gtid_cap, ((*output_index, Err(err)), new_gtid.clone(), 1))
-                        .await;
-                    ctx.errored_tables.insert(dropped_table.clone());
+                    if let Some((output_index, _)) = ctx.table_info.get(dropped_table) {
+                        let gtid_cap = ctx.data_cap_set.delayed(new_gtid);
+                        ctx.data_output
+                            .give(&gtid_cap, ((*output_index, Err(err)), new_gtid.clone(), 1))
+                            .await;
+                        ctx.errored_tables.insert(dropped_table.clone());
+                    }
```

Comment:
this diff was just to fix a rust-analyzer error complaining about
```
                    let (output_index, _) = &ctx.table_info[&dropped_table];
```
not properly matching all values